language/samples/v1/samplegen_map_field_access.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# DO NOT EDIT! This is a generated sample ("Request",  "samplegen_map_field_access")
+
+# To install the latest published package dependency, execute the following:
+#   pip install google-cloud-language
+
+# sample-metadata
+#   title: This sample reads and loops over a map field in the response
+#   description: This sample reads and loops over a map field in the response
+#   usage: python3 samples/v1/samplegen_map_field_access.py
+
+# [START samplegen_map_field_access]
+from google.cloud import language_v1
+from google.cloud.language_v1 import enums
+
+
+def sample_analyze_entities():
+    """This sample reads and loops over a map field in the response"""
+
+    client = language_v1.LanguageServiceClient()
+
+    type_ = enums.Document.Type.PLAIN_TEXT
```

Comment:
Sweet suggestion!

Thus far, when authoring these samples, I've often added these comments manually _(which you can do via the configuration)_ but have't looked into generating links with the generator.

We could presumably construct something close to a permalink, given any type name, but we wouldn't be able to include the version number of the pip package, so the link would 404 if the type were ever removed.

^--- synth could add the version number to these links, though.

@busunkim96 Any idea if the Python documentation includes the available enum values with descriptions?

I found this page, but it doesn't include the enum values: https://googleapis.dev/python/language/latest/gapic/v1/types.html

Reply:
Bump / ping @busunkim96 

Wondering if the generated Python library documentation includes enums anywhere?

@kurtisvg The alternative (_not ideal but we've done this_) is putting the responsibility for adding a link on the sample author + having them link to the enum options in the API reference

This [published sample](https://cloud.google.com/natural-language/docs/analyzing-sentiment#language-sentiment-string-python) lists all available types, but it comes from this [manually authored sample config line](https://github.com/googleapis/googleapis/blob/617d4430e337c82a8c99a8cb9f3d9e16e23ab22d/google/cloud/language/v1/samples/language_syntax_text.yaml#L16). Not ideal. But at least it's 1 canonical source for this sample in all languages.

Reply:
I'm okay with manually adjusting for now - presumable these auto-gen samples will still get reviewed like normal ones, so we can encourage it that way if it's listed in the sample guidelines.

Reply:
> @busunkim96 Any idea if the Python documentation includes the available enum values with descriptions?

This is https://github.com/googleapis/artman/issues/579 / https://github.com/googleapis/google-cloud-python/issues/7503

Reply:
I wonder... since the generators have the .proto, maybe including the list in the samples _is_ the right way to go. We automatically add a comment which lists the available options. No URL required (_but great if possible to construct_).

@software-dov Thoughts on grabbing the enum display names in the sample generator and automatically including them?

**This is language agnostic but seems to be one of the most popular pieces of feedback!** 🎉